src/main/java/cz/cvut/kbss/analysis/config/SecurityConfig.java

```diff
@@ -49,20 +55,22 @@ CorsConfigurationSource corsConfigurationSource() {
         return source;
     }
 
-    @Override
-    protected void configure(HttpSecurity http) throws Exception {
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        log.debug("Using internal security mechanisms.");
+        final AuthenticationManager authManager = buildAuthenticationManager(http);
         http
-                .cors()
-                .and()
-                .httpBasic().disable()
-                .csrf().disable()
-                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS)
-                .and()
-                .authorizeRequests()
-                .antMatchers("/auth/register").permitAll()
-                .antMatchers("/auth/signin").permitAll()
-                .anyRequest().authenticated()
-                .and()
+//                .requestMatcher(r -> false)
+                .cors(auth -> auth.configurationSource(corsConfigurationSource()))
+                .httpBasic(basic -> basic.disable())
+                .csrf(csrf -> csrf.disable())
+                .sessionManagement(sessionManagement -> sessionManagement.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
+                .authorizeHttpRequests(auth -> auth
+                        .requestMatchers(AntPathRequestMatcher.antMatcher("/auth/register")).permitAll()
+                        .requestMatchers(AntPathRequestMatcher.antMatcher("/auth/signin")).permitAll()
+                        .anyRequest().authenticated())
```

Comment:
I think there should be something like `authenticationManager(buildAuthenticationManager(http))`. But, my changes in the [PR](https://github.com/kbss-cvut/fta-fmea/pull/50) affect this anyway, so it doesn't really matter.

Reply:
For it works without calling`authenticationManager(buildAuthenticationManager(http))` on the filter chain. It seams that the authentication provider bean is automatically set up in the filter chain. 

Also, I removed the line with the unused `authManager` variable :
`final AuthenticationManager authManager = buildAuthenticationManager(http);`